File: CostAccounting.Web.Angular/ClientApp/src/app/core/models/expense.ts
import { Category } from "./category";

export class Expense {
    id: number;
    amount: number;
    category: Category;
    categoryId: string;
    date: Date;

    constructor(data: any) {
        Object.assign(this, data);

        this.date = new Date(data.date);
    }
}

File: src/Domain/Entities/Expense.cs
using System;
using Domain.Common;

namespace Domain.Entities
{
    public class Expense : Entity<long>
    {
        public Guid CategoryId { get; set; }
        public Guid UserId { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }

        public Category Category { get; set; }
        public User User { get; set; }
    }
}
